src/Network/OAuth/OAuth2/Internal.hs

```diff
@@ -63,12 +67,62 @@ instance FromJSON OAuth2Token where
 instance ToJSON OAuth2Token where
     toEncoding = genericToEncoding defaultOptions { fieldLabelModifier = camelTo2 '_' }
 
+-- | Possible OAuth error types
+-- | Not all of them can appear everywhere, but this is good enough for now.
```

Comment:
... now I remember why I didn't do that. You can't have two error codes with the same name. WA by using a module?